src/vs/workbench/parts/terminal/electron-browser/terminalActions.ts

```diff
@@ -227,6 +228,64 @@ export class MoveToLineEndTerminalAction extends BaseSendTextTerminalAction {
 	}
 }
 
+export class SendSequenceTerminalCommand extends Command {
+	public static readonly ID = TERMINAL_COMMAND_ID.SEND_SEQUENCE;
+	public static readonly LABEL = nls.localize('workbench.action.terminal.sendSequence', "Send Custom Sequence To Terminal");
+
+	public runCommand(accessor: ServicesAccessor, args: any): void {
+		const terminalInstance = accessor.get(ITerminalService).getActiveInstance();
+		if (!terminalInstance) {
+			return;
+		}
+		const text = args.text.split(' ');
+		var terminalText;
+		for (let i = 0; i < text.length; i++) {
+			terminalText = ' ' + prepareTerminaText(text[i]);
+			terminalInstance.sendText(terminalText, false);
+		}
+	}
+}
+
+function prepareTerminaText(text: string): string {
+	// Hexcodes and excape char
+	if (text.substring(0, 2).toLowerCase() === '\\x') {
```

Comment:
This needs to happen for every character, I suggest an algorithm like this:

```
for each char
  detect if it's hex. if so translate and "consume" the characters and move the loop to the following character
```

Reply:
By `each character`, do you mean `each space separated word`? If so, this function is being called for `each space separated word`.

Reply:
I think we have different ideas for the end solution, you want to pass in a space separated list of characters or words that translate to characters, I think the most flexible way would be to pass in a list of characters. So instead of this:

```
"args": {"text": "ctrl+c up enter"}
```

 It would be:

```
"args": {"text": "\x03\x1b[A\x0d"}
```

I think we should start with this as it supports everything and then see if we need something better. In fact if we support my proposal an extension could provide the nicer interface if needed.

Reply:
I could see something like this being nice:

```json
"args": {
  "keys": [
    "ctrl+c",
    "up",
    "enter"
  ]
}
```

But I think it should live in an extension first to see if people find it useful.

Reply:
On implementation side, implementing this is easy:
`
"args": {"text": "\x03\x1b[A\x0d"}
`
`
For each char, termnalInstance.sendText(String.fromChar(char))
`
But will it be easy enough for end user to use?
Final decision is yours. Let me know what way should we go :)
BTW, How should we handle `[A` if we go that way?

Reply:
I think this is a good first step as it allows you to do everything. We can evaluate after if we need to do more. A section on https://code.visualstudio.com/docs/editor/integrated-terminal might be enough to cover this with some mappings between common keys and their sequences since it will need documentation for discoverability anyway.

It should act like a JS string, `"\x03"` might just work (as opposed to `"\\x03"`). Maybe the string doesn't need to be changed at all? In other words, `{"text": "\x03\x1b[A\x0d"}` should act exactly the same as `{"text": "\x03\x1b\x5b\x41\x0d"}`

Reply:
Some final clarifications, Tell me if that looks cool :)
1. I'll have to deal with hex char sequence and nothings else (Removing `prepareTerminalText` function will make it simpler, just use `for each char` in `runCommand() itself`). That is strings like `"\x03\x1b\x5b\x41\x0d"`.

2. `text` does not contain any spaces. If we wish to allow spaces, that's fine as well.

3. Using `\x03`  in `keybinding.json` file gives error. To be specific, `[jsonc] Invalid escape character in string.`

Reply:
Sorry about the delay, had too many notifications 😱 

1. Yes, I would also expect `\a`, `\b`, etc. to work as specified in this file: https://github.com/xtermjs/xterm.js/blob/f47f095b8928a6e50ce80f87d7fd11e50cf9a190/src/common/data/EscapeSequences.ts#L25, not sure if they will without special casing
2. text should allow spaces, using `{ "text": "hello world" }` should send that, including the space
3. We probably need to use `\\x03` in the keybindings.json

Reply:
I just tested sending directly to the process in the xterm.js demo (what VS Code's terminal is based on) and sending `\x03` definitely doesn't eat the next character for me:

<img width="976" alt="screen shot 2018-08-31 at 9 27 45 am" src="https://user-images.githubusercontent.com/2193314/44924443-2bb9c400-ad00-11e8-9713-0697adc39bf9.png">

Reply:
![screenshot_2018-08-31_23-22-34](https://user-images.githubusercontent.com/15014949/44928224-d9ff5100-ad74-11e8-9780-94dd9ba0bba2.png)

Can you please try `term._core.handler("\x03ls\n")`

Reply:
I can repro this in Linux, I'm guessing this is expected behavior of bash. For this I would try workaround it by using something like `\r` to make sure the ^C goes though which apppears to work`\x03\rls\n`, it adds another line but that's not a big deal.

![image](https://user-images.githubusercontent.com/2193314/44929864-506a6780-ad11-11e8-9cb2-48ce7df41820.png)